minstack.py

```diff
@@ -0,0 +1,72 @@
+#Time Complexity - push(): O(1)
+# pop():O(1) 
+# top() : O(1)
+# getMin() : 
+#Space Complexity - Overall it's O(n) where n are total inputs but all function implementations are O(1)
+
+# Running on leetcode : Yes
+#Single stack solution for Minstack - similar complexity 
+class MinStack2:
+
+    def __init__(self):
+        """
+        initialize your data structure here.
+        """
+        self.stack = []
+        self.min = float('inf')
+        
+    def push(self, x: int) -> None:
+        if x<= self.min:
+            self.stack.append(self.min)
+            self.min = x 
+        self.stack.append(x)
+
+    def pop(self) -> None:
+        val = self.stack.pop()
+        if self.min==val:
+            newmin = self.stack.pop()
+            self.min = newmin
+
+    def top(self) -> int:
+        return self.stack[-1]
```

Comment:
for pop and top, please add edge cases to prevent from operating on an empty stack.